pkg/lexer/lexer.go

```diff
@@ -0,0 +1,84 @@
+package lexer
+
+type TokenType int
+
+const (
+	// Special tokens
+	ILLEGAL TokenType = iota
+	EOF
+	SEMICOLON // ;
+	NEWLINE   // \n
+
+	// Identifiers and literals
+	IDENT
+
+	// Logical operators
+	AND
+	OR
+
+	// Quote operators
+	SINGLE_QUOTE
+	DOUBLE_QUOTE
+
+	// Redirection operators
+	TO_RIGHT
+	TO_LEFT
+
+	// Pipe operators
+	PIPE
+)
+
+func (t TokenType) String() string {
+	var tokenMap = map[TokenType]string{
+		EOF:          "EOF",
+		SEMICOLON:    ";",
+		NEWLINE:      "\n",
+		IDENT:        "IDENT",
+		AND:          "&&",
+		OR:           "||",
+		SINGLE_QUOTE: "'",
+		DOUBLE_QUOTE: "\"",
+		TO_RIGHT:     ">",
+		TO_LEFT:      "<",
+		PIPE:         "|",
+	}
+
+	return tokenMap[t]
+}
+
+type Token struct {
+	Type  TokenType
+	Value string
```

Comment:
Add a position field, which will be useful for error reporting.

Reply:
That information is along with lexer.Tokens this is the type token and there can be many of the same type so pos here doesn't make sense

Reply:
The tokens will finally be handled by the parser. How do you want to report where the error occurs when there is a parse error?

Reply:
The oh lol, yeah I see your point now. Can add that easily

Reply:
I haven't added it yet but the lexer is working now, can you take a look at what I've done so far and see if it makes sense. I will add this later once I have a working REPL.